spring-components/rdf4j-spring/src/main/java/org/eclipse/rdf4j/spring/support/DataInserter.java

```diff
@@ -41,7 +42,18 @@ public void insertData(Resource dataFile) {
 		logger.debug("Loading data from {}...", dataFile);
 		try {
 			RepositoryConnection con = connectionFactory.getConnection();
-			con.add(dataFile.getInputStream(), "", RDFFormat.TURTLE);
+			RDFFormat fmt = RDFFormat.matchFileName(
+					dataFile.getFilename(),
+					Set.of(
+							RDFFormat.TURTLE,
+							RDFFormat.TRIG,
+							RDFFormat.NTRIPLES,
+							RDFFormat.NQUADS))
+					.orElseThrow(
+							() -> new IllegalArgumentException(
+									"Failed to determine file format of input file "
+											+ dataFile));
+			con.add(dataFile.getInputStream(), "", fmt);
```

Comment:
I'd change this to 

```java
RDFFormat fmt = Rio.getParserFormatForFilename(dataFile.getName()).orElseThrow(....)
```
Advantage is that this uses all registered parsers to determine a suitable format, instead of a hard-coded list.

Reply:
ah, yes. Thanks.